lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -167,9 +167,8 @@ size_t firrtl::getNumPorts(Operation *op) {
 /// Check whether an operation has a `DontTouch` annotation, or a symbol that
 /// should prevent certain types of canonicalizations.
 bool firrtl::hasDontTouch(Operation *op) {
-  if (isa<FExtModuleOp, FModuleOp>(op))
-    return AnnotationSet(op).hasDontTouch();
-  return op->getAttr(hw::InnerName::getInnerNameAttrName()) != nullptr;
+  return op->getAttr(hw::InnerName::getInnerNameAttrName()) ||
+         AnnotationSet(op).hasDontTouch();
```

Comment:
Especially since symbols are now no longer "dontTouch", can we separate these?  Are there users expecting the "hasDontTouch" predicate to answer "or has a symbol"?  If so, maybe we can rename this?

Reply:
I think conventionally we have used a term "dontTouch" to represent an optimization blocking property and inner symbols still imply "dontTouch".